_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5084,30 +5084,100 @@ mfxU64 make_back_color_yuv(mfxU16 bit_depth, mfxU16 Y, mfxU16 U, mfxU16 V)
         ;
-};
+}
 
 inline
-mfxU64 make_def_back_color_yuv(mfxU16 bit_depth)
+mfxU64 make_back_color_argb(mfxU16 bit_depth, mfxU16 R, mfxU16 G, mfxU16 B)
 {
-    assert(bit_depth);
+    VM_ASSERT(bit_depth >=8);
+
+    mfxU64 const max_val = 1 << bit_depth;
 
+    return ((mfxU64) (max_val - 1) << 48) |
+           ((mfxU64) VPP_RANGE_CLIP(R, 0, max_val - 1) << 32) |
+           ((mfxU64) VPP_RANGE_CLIP(G, 0, max_val - 1) << 16) |
+           ((mfxU64) VPP_RANGE_CLIP(B, 0, max_val - 1) << 0);
+}
+
+inline
+mfxU64 make_def_back_color_yuv(mfxU16 bit_depth)
+{
+    VM_ASSERT(bit_depth >= 8);
     mfxU64 const shift = bit_depth - 8;
-    mfxU64 const min_val = 16 << shift, max_val = 256 << shift;
-    return
-        ((max_val - 1ULL) << 48) |
-        ( min_val      << 32) |
-        ( max_val / 2  << 16) |
-        ( max_val / 2  <<  0)
-        ;
-};
+    mfxU16 const min_val = 16 << shift, max_val = 256 << shift;
+    return make_back_color_yuv(bit_depth, min_val, max_val / 2, max_val / 2);
+}
+
+inline
+mfxU64 make_def_back_color_argb(mfxU16 bit_depth)
+{
+    return make_back_color_argb(bit_depth, 0, 0, 0);
+}
+
+static
+mfxU64 get_background_color(const mfxVideoParam & videoParam)
+{
+    for (mfxU32 i = 0; i < videoParam.NumExtParam; i++)
+    {
+        if (videoParam.ExtParam[i]->BufferId == MFX_EXTBUFF_VPP_COMPOSITE)
+        {
+            mfxExtVPPComposite *extComp = (mfxExtVPPComposite *) videoParam.ExtParam[i];
+            switch (videoParam.vpp.Out.FourCC) {
+                case MFX_FOURCC_NV12:
+                case MFX_FOURCC_YV12:
+                case MFX_FOURCC_NV16:
+                case MFX_FOURCC_YUY2:
+                case MFX_FOURCC_AYUV:
+                    return make_back_color_yuv(8, extComp->Y, extComp->U, extComp->V);
+                case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+                case MFX_FOURCC_Y210:
+                case MFX_FOURCC_Y410:
+#endif
+                case MFX_FOURCC_P210:
+                    return make_back_color_yuv(10, extComp->Y, extComp->U, extComp->V);
+                case MFX_FOURCC_RGB4:
+                case MFX_FOURCC_BGR4:
+                    return make_back_color_argb(8, extComp->R, extComp->G, extComp->B);
+                default:
+                    break;
+            }
+        }
+    }
+
+    switch (videoParam.vpp.Out.FourCC)
+    {
+        case MFX_FOURCC_NV12:
+        case MFX_FOURCC_YV12:
+        case MFX_FOURCC_NV16:
+        case MFX_FOURCC_YUY2:
+        case MFX_FOURCC_AYUV:
+            return make_def_back_color_yuv(8);
+        case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y210:
+        case MFX_FOURCC_Y410:
+#endif
+        case MFX_FOURCC_P210:
+            return make_def_back_color_yuv(10);
+        case MFX_FOURCC_RGB4:
+        case MFX_FOURCC_BGR4:
+            return make_def_back_color_argb(8);
+        default:
+            break;
+    }
+
+    VM_ASSERT(0);
```

Comment:
Ok. This patch looks almost good except errors handling. We will get unpredictable behavior on new color formats not handled here. Thus, please, change VM_ASSERTs to c++ throw of exceptions. We probably want to have runtime error (even very generic one) rather than silent behavior with wrong color format.

This applies to all VM_ASSERTs including bit_depth ones above.